cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/DefaultHttpServiceResponder.java

```diff
@@ -154,10 +152,20 @@ public void sendError(int status, String errorMessage) {
    * @param headers the headers to be sent back
    */
   @Override
-  public void send(int status, ByteBuffer content, String contentType, Multimap<String, String> headers) {
-    responder.sendContent(HttpResponseStatus.valueOf(status),
-                          ChannelBuffers.wrappedBuffer(content), contentType, headers);
-    emitMetrics(status);
+  public void send(final int status, ByteBuffer content, final String contentType, Multimap<String, String> headers) {
+    bufferedResponse = new BufferedResponse(status, content, contentType, headers);
```

Comment:
What I meant is that if the user's handler calls send twice, then that is a bug in the user code. Previously, that would have caused exceptions in the log. Now it would go unnoticed. I think it is better to make the developer aware of the bugs in his code.